blocks/library/embed/index.js

```diff
@@ -73,52 +72,130 @@ registerBlock( 'core/embed', {
+				fetching: false,
+			};
+			this.noPreview = [
+				'facebook.com',
+			];
+			if ( this.props.attributes.url ) {
+				// if the url is already there, we're loading a saved block, so we need to render
+				this.doServerSideRender();
+			}
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		componentWillUnmount() {
+			// can't abort the fetch promise, so let it know we will unmount
+			this.unmounting = true;
+		}
+
+		doServerSideRender( event ) {
+			if ( event ) {
+				event.preventDefault();
+			}
+			const { url } = this.props.attributes;
+			const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce;
+
+			this.setState( { error: false, fetching: true } );
+			fetch( api_url, {
+				credentials: 'include',
+			} ).then(
+				( response ) => {
+					if ( this.unmounting ) {
+						return;
+					}
+					response.json().then( ( obj ) => {
+						const { html, type } = obj;
+						if ( html ) {
+							this.setState( { html, type } );
+						} else {
+							this.setState( { error: true } );
+						}
+						this.setState( { fetching: false } );
+					} );
+				}
 			);
 		}
 
-		return (
-			<figure className="blocks-embed">
-				<div className="iframe-overlay">
-					<iframe src={ url } title={ title } />
-				</div>
-				{ ( caption && caption.length > 0 ) || !! focus ? (
-					<Editable
-						tagName="figcaption"
-						placeholder={ wp.i18n.__( 'Write caption…' ) }
-						value={ caption }
-						focus={ focus }
-						onFocus={ setFocus }
-						onChange={ ( value ) => setAttributes( { caption: value } ) }
-						inline
-						inlineToolbar
-					/>
-				) : null }
-			</figure>
-		);
+		render() {
+			const { html, type, error, fetching } = this.state;
+			const { url, caption } = this.props.attributes;
+			const { setAttributes, focus, setFocus } = this.props;
+
+			if ( ! html ) {
+				return (
+					<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
+						<form onSubmit={ this.doServerSideRender }>
+							<input
+								type="url"
+								className="components-placeholder__input"
+								placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) }
+								onChange={ ( event ) => setAttributes( { url: event.target.value } ) } />
+							{ ! fetching
+								? <Button
+									isLarge
+									type="submit">
+									{ wp.i18n.__( 'Embed' ) }
+								</Button>
+								: <Spinner />
+							}
+							{ error && <p className="components-placeholder__error">{ wp.i18n.__( 'Sorry, we could not embed that content.' ) }</p> }
+						</form>
+					</Placeholder>
+				);
+			}
+
+			const domain = url.split( '/' )[ 2 ].replace( /^www\./, '' );
```

Comment:
yeah, the host without www. because we want to blacklist certain embeds from previewing in the editor (facebook specifically). I'll look at the url module :)